### ISSUE TITLE
Improve handling of unexpected situations

### ISSUE THREAD

**Issue by mudrd8mz** [SVIeSolutions/moodle-mod_via#1]
When trying to add an instance of the module with the API key left empty, I am getting following errors:

```
Notice: Undefined variable: response in /mod/via/locallib.php on line 482
Warning: Invalid argument supplied for foreach() in /mod/via/locallib.php on line 482
```

The try/catch block in the `via_get_list_profils()` does not seem to handle the situation well.

**Comment by SVIeSolutions** [comment]
If you don’t have an API key then you can’t connect to Via and that’s why you get errors…
Should I display something else? Maybe that you need an API Key to continue.

Would you like me to make a presentation of the tool online?

---

Alexandra Dinan-Mitchell
Designer multimédia | Analyste-programmeur
Multimedia designer | Programmer analyst

[Image supprimée par l'expéditeur. SVI eSolutions - Découvrez Via eLearning & eMeeting]http://www.sviesolutions.com/

[Image supprimée par l'expéditeur. TC)lC)phone:]418 948.4848 #223 | 866 843.4848 #223
[Image supprimée par l'expéditeur. Courriel:]alexandra.dinan@sviesolutions.commailto:alexandra.dinan@sviesolutions.com

[Image supprimée par l'expéditeur. Twitter]http://twitter.com/sviesolutions[Image supprimée par l'expéditeur. Facebook]http://www.facebook.com/pages/SVI-eSolutions-inc/161880277164953[Image supprimée par l'expéditeur. LinkedIn]http://www.linkedin.com/company/434174

sviesolutions.comhttp://www.sviesolutions.com/ | assistance.sviesolutions.comhttps://assistance.sviesolutions.com/

De : David Mudrák [mailto:notifications@github.com]
Envoyé : 24 février 2015 04:48
À : SVIeSolutions/moodle-mod_via
Objet : [moodle-mod_via] Improve handling of unexpected situations (#1)

When trying to add an instance of the module with the API key left empty, I am getting following errors:

Notice: Undefined variable: response in /mod/via/locallib.php on line 482

Warning: Invalid argument supplied for foreach() in /mod/via/locallib.php on line 482

The try/catch block in the via_get_list_profils() does not seem to handle the situation well.

**Comment by mudrd8mz** [comment]
I do understand why it happens. What I was trying to say is that your code should detect the empty API key before trying to contact the server and iterate over non-existing response array. In other words, that PHP notice and warning should be never displayed to users.

**Comment by SVIeSolutions** [comment]
Yes, I have made the modifications and will update the plugin shortly.
I want to look at your other comment about the ‘sesskey’ before adding the new version.

Thank you for your help and let meknow if you would like a demo! We have a great tool!

---

Alexandra Dinan-Mitchell
Designer multimédia | Analyste-programmeur
Multimedia designer | Programmer analyst

[SVI eSolutions - Découvrez Via eLearning & eMeeting]http://www.sviesolutions.com/

[TC)lC)phone:]418 948.4848 #223 | 866 843.4848 #223
[Courriel:]alexandra.dinan@sviesolutions.commailto:alexandra.dinan@sviesolutions.com

[Twitter]http://twitter.com/sviesolutions[Facebook]http://www.facebook.com/pages/SVI-eSolutions-inc/161880277164953[LinkedIn]http://www.linkedin.com/company/434174

sviesolutions.comhttp://www.sviesolutions.com/ | assistance.sviesolutions.comhttps://assistance.sviesolutions.com/

De : David Mudrák [mailto:notifications@github.com]
Envoyé : 24 février 2015 15:48
À : SVIeSolutions/moodle-mod_via
Cc : Alexandra Dinan-Mitchell
Objet : Re: [moodle-mod_via] Improve handling of unexpected situations (#1)

I do understand why it happens. What I was trying to say is that your code should detect the empty API key before trying to contact the server and iterate over non-existing response array. In other words, that PHP notice and warning should be never displayed to users.

**Comment by mudrd8mz** [comment]
> We have a great tool!

Why not to make a short demo video on YouTube that you could then embed into your plugin description at the moodle.org/plugins? :-)
